### PR TITLE
Case 20882: Support custom display / input plugin lists and small tweaks

### PR DESCRIPTION
**Pull request by jherico**
[20882](https://highfidelity.manuscript.com/f/cases/20882/Support-CMake-time-overrides-of-normal-display-plugins)

* Updates the CMake configuration and the `input-plugins` and `display-plugins` libraries to allow applications to suppress the default built-in plugins and supply their own (necessary for Oculus Mobile and potential other future standalone HMDs)
* Expose some critical OpenGL ES plugin availability to the code via preprocessor macros
* Ensure GLES backends will use the instanced stereo drawing technique if `EXT_clip_cull_distance` is available
* Merge in updates to the `oculusMobile` and `oculusMobilePlugin` libraries from the feature/quest branch
* Ensure the QML file selector is exposed at the `hifi::qml::OffscreenSurface` level
* Remove use of `FileLogger` in android builds (only logcat should be used, not file based logging)

## Testing

* Verify file logging works correctly on desktop builds.
* Verify Android startup and shutdown work the same as in master builds
* Verify Android rendering is correct
* Verify the list of display plugins is unchanged from master
* Verify web content (web entities, marketplace) works from Desktop builds.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17131/HighFidelity-Beta-Interface-PR14875-f474cbd.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17131/HighFidelity-Beta-PR14875-f474cbd.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17131/HighFidelity-Beta-Interface-PR14875-f474cbd.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17143/HighFidelity-Beta-PR14875-d330180.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17143/HighFidelity-Beta-PR14875-d330180.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17143/HighFidelity-Beta-Interface-PR14875-d330180.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17143/HighFidelity-Beta-PR14875-d330180.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17143/HighFidelity-Beta-Interface-PR14875-d330180.exe

**Comment by JeffClinton** [comment]
Currently testing

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17202/HighFidelity-Beta-PR14875-b896e25.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17202/HighFidelity-Beta-PR14875-b896e25.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17202/HighFidelity-Beta-Interface-PR14875-b896e25.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17202/HighFidelity-Beta-PR14875-b896e25.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17202/HighFidelity-Beta-Interface-PR14875-b896e25.exe

File: android/libraries/qt/src/main/java/org/qtproject/qt5/android/bindings/QtActivity.java [hunk 1]
```diff
@@ -650,6 +650,8 @@ protected void onStop() {
         if (!keepInterfaceRunning) {
             QtApplication.invokeDelegate();
         }
+        QtNative.terminateQt();
```
Comment (amerhifi) [inline android/libraries/qt/src/main/java/org/qtproject/qt5/android/bindings/QtActivity.java]: These aren't necessary anymore. (As well as the keepInterfaceRunning) but we can fix in the lifecycle PR

**Comment by samcake** [comment]
THis pr has been tested by devs on the second pass after initial feedback from qa.
ready to be merged  and tested in integration

**Pull request merged**